### PR TITLE
docs: how to customise the php configuration

### PR DESCRIPTION
**Pull request by Gaitholabi**
The purpose of this pull request is to allow setting custom PHP configuration.

The users can customize their PHP by setting the custom  [ini fields](https://www.php.net/manual/en/configuration.file.php
). This can be achieved by creating `*/php/custom-config.ini` file in the relevant template.

The following directives cannot be customized before 3.6.0 is released:
- expose_php - default is `off`
- memory_limit - default is 256M

### CODE REVIEW

File: templates/aws/eb/HA/cron/php/custom-config.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#add your custom config here
```
Comment (bytestream) [inline templates/aws/eb/HA/cron/php/custom-config.ini:1]:
```suggestion
# add your custom config here
```

File: templates/aws/eb/HA/web/php/custom-config.ini [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#add your custom config here
```
Comment (bytestream) [inline templates/aws/eb/HA/web/php/custom-config.ini:1]:
```suggestion
# add your custom config here
```

File: templates/aws/eb/Preview/php/custom-config.ini [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#add your custom config here
```
Comment (bytestream) [inline templates/aws/eb/Preview/php/custom-config.ini:1]:
```suggestion
# add your custom config here
```

File: templates/aws/eb/Single/php/custom-config.ini [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#add your custom config here
```
Comment (bytestream) [inline templates/aws/eb/Single/php/custom-config.ini:1]:
```suggestion
# add your custom config here
```

File: templates/docker-compose/php/custom-config.ini [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#add your custom config here
```
Comment (bytestream) [inline templates/docker-compose/php/custom-config.ini:1]:
```suggestion
# add your custom config here
```

**Review by bytestream — COMMENTED** [review]
`custom-config.ini` shoudn't be tracked by git should it?

**Comment by Gaitholabi** [comment]
> `custom-config.ini` shoudn't be tracked by git should it? I think it should be in `configs/php/custom.ini` and then copied like we do with the nginx templates?

I think that keeping it and ignoring `/php/*` feels like easier to manage. It's one less step in the configuration, and we won't be adding any custom property across all deployments ourselves, and if so, we'll be doing that internally.

so IMO, having it here, without the need for users to do any copying, since it will always be there, feels easier

File: templates/docker-compose/docker-compose.prod.yml.dist [hunk 6]
```diff
@@ -13,6 +13,7 @@ services:
             - supportpal_config:/supportpal/config/production
             - supportpal_storage:/supportpal/storage
             - ./customization/:/customization
+            - ./php/custom.ini:/usr/local/etc/php/9999-custom-config.ini
```
Comment (bytestream) [inline templates/docker-compose/docker-compose.prod.yml.dist:16]: I think the mapped path should be `/user/local/etc/php/conf.d/9999-custom-config.ini`

```
# php -i | more
phpinfo()
PHP Version => 7.4.16

System => Linux c3143b42510a 5.8.0-59-generic #66~20.04.1-Ubuntu SMP Thu Jun 17 11:14:10 UTC 2021 x86_64
Build Date => Apr 10 2021 13:47:26
Configure Command =>  './configure'  '--build=x86_64-linux-gnu' '--with-config-file-path=/usr/local/etc/php' '--with-config-file-scan-dir=/usr/local/etc/php/conf.d' '--enable-option-checking=fatal' '--with-mhash' '--with-pi
c' '--enable-ftp' '--enable-mbstring' '--enable-mysqlnd' '--with-password-argon2' '--with-sodium=shared' '--with-pdo-sqlite=/usr' '--with-sqlite3=/usr' '--with-curl' '--with-libedit' '--with-openssl' '--with-zlib' '--with-p
ear' '--with-libdir=lib/x86_64-linux-gnu' '--enable-fpm' '--with-fpm-user=www-data' '--with-fpm-group=www-data' '--disable-cgi' 'build_alias=x86_64-linux-gnu'
Server API => Command Line Interface
Virtual Directory Support => disabled
Configuration File (php.ini) Path => /usr/local/etc/php
Loaded Configuration File => (none)
Scan this dir for additional .ini files => /usr/local/etc/php/conf.d

```

**Pull request merged**